core/services/ocr2/delegate.go

```diff
@@ -590,9 +590,22 @@ func (d *Delegate) newServicesGenericPlugin(
 	}
 
 	errorLog := &errorLog{jobID: jb.ID, recordError: d.jobORM.RecordError}
+	var providerClientConn grpc.ClientConnInterface
 	providerConn, ok := provider.(connProvider)
-	if !ok {
-		return nil, errors.New("provider not supported: the provider is not a LOOPP provider")
+	if ok {
+		providerClientConn = providerConn.ClientConn()
+	} else {
+		d.lggr.Info("provider is not a LOOPP provider, switching to provider server")
+
+		ps, err := relay.NewProviderServer(provider, types.OCR2PluginType(cconf.ProviderType), d.lggr)
```

Comment:
Please document why we're dealing with this difference here vs in the NewServerAdapter struct (which could be argued to be a better place)